client/src/controllers/SkipLinkController.ts

```diff
@@ -0,0 +1,33 @@
+import { Controller } from '@hotwired/stimulus';
+
+/**
+ * Appears at the top left corner of the admin page with the tab button is clicked.
+ * Used to provide an accessible skip button for keyboard control so that users can
+ * easily navigate to the main content without having to navigate a long list of navigation links.
+ *
+ * Inspired by https://github.com/selfthinker/dokuwiki_template_writr/blob/master/js/skip-link-focus-fix.js
+ */
+export class SkipLinkController extends Controller<HTMLAnchorElement> {
+  skipToTarget?: HTMLElement | null;
```

Comment:
>   skipToTarget?: HTMLElement | null;

1. `skipToTarget` is a property of the class instance `SkipLinkController`
2. `?` - this could be undefined (aka optional) this will be the case in the short time between the class creation & the `connect` method being called. This may only be for a few ms but it still can happen.
3. `HTMLElement | null` - this says, when we DO assign something to the property it will probably be a `HTMLElement` but could still be `null`. This is because when we DO assign something we use `document.querySelector` which MAY return `null`.